digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -1771,12 +1775,12 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
     val floatingValue = source.isDefined && source.get.validTo.isDefined && source.get.validTo.get.isBeforeNow
     val roadAddress = RoadAddress(source.map(_.id).getOrElse(NewRoadAddress), pl.roadNumber, pl.roadPartNumber, pl.roadType, pl.track, pl.discontinuity,
       pl.startAddrMValue, pl.endAddrMValue, None, None, pl.createdBy, pl.linkId, pl.startMValue, pl.endMValue, pl.sideCode,
-      pl.linkGeometryTimeStamp, pl.toCalibrationPoints(), floating = false, geom, pl.linkGeomSource, pl.ely, terminated = NoTermination, source.map(_.commonHistoryId).getOrElse(0))
+      pl.linkGeometryTimeStamp, pl.toCalibrationPoints(), floating = NoFloating, geom, pl.linkGeomSource, pl.ely, terminated = NoTermination, source.map(_.commonHistoryId).getOrElse(0))
     pl.status match {
       case UnChanged =>
-        roadAddress.copy(startDate = source.get.startDate, endDate = source.get.endDate, floating = floatingValue)
+        roadAddress.copy(startDate = source.get.startDate, endDate = source.get.endDate, floating = if (floatingValue) FloatingReason.GeometryChanged else NoFloating) // TODO Floating reason? - Sami
       case Transfer | Numbering =>
-        roadAddress.copy(startDate = Some(project.startDate), floating = floatingValue)
+        roadAddress.copy(startDate = Some(project.startDate), floating = if (floatingValue) FloatingReason.NewAddressGiven else NoFloating) // TODO Floating reason? - Sami
```

Comment:
What should be the floating reason in this case?

Reply:
I think the reason being NewAddressGiven is not that bad.
Floating through:
- ApplyChanges batch -> reason: ApplyChanges
- Find Floating batch -> reason: GeometryChanged or NewAddressGiven, or GapInGeometry 
- Manualy by user in UI -> reason: ManualFloating 

So i would say that we could use that NewAddressGiven since in fact we kind of create one new address road with spliting tool. Or do you think it would be better to have one new type to know were it was done through SplitingTool?